fga/cache/cachedFgaTenantStore_test.go

```diff
@@ -0,0 +1,173 @@
+				cachedStore.GetCache().Add("tenant-tenant123", storeId)
+
+				client.EXPECT().
+					ReadAuthorizationModels(ctx, &openfgav1.ReadAuthorizationModelsRequest{StoreId: "store123"}).
+					Return(&openfgav1.ReadAuthorizationModelsResponse{
+						AuthorizationModels: []*openfgav1.AuthorizationModel{
+							{Id: modelId},
+						}}, nil).
+					Once()
+			},
+			expectedModelID: modelId,
+			expectedError:   nil,
+		},
+		{
+			name: "ListStores_Error",
+			setupMock: func(client *mocks.OpenFGAServiceClient, cachedStore *CachedFgaTenantStore) {
+				client.EXPECT().
+					ListStores(ctx, &openfgav1.ListStoresRequest{}).
+					Return(nil, assert.AnError).
+					Once()
+			},
+			expectedError: assert.AnError,
+		},
+		{
+			name: "MatchingKeyNotFound_Error",
+			setupMock: func(client *mocks.OpenFGAServiceClient, cachedStore *CachedFgaTenantStore) {
+				client.EXPECT().
+					ListStores(ctx, &openfgav1.ListStoresRequest{}).
+					Return(&openfgav1.ListStoresResponse{Stores: []*openfgav1.Store{}}, nil).
+					Once()
+			},
+			expectedError: errors.New("could not find store matching key \"tenant-tenant123\""),
+		},
+		{
+			name: "ReadAuthorizationModels_Error",
+			setupMock: func(client *mocks.OpenFGAServiceClient, cachedStore *CachedFgaTenantStore) {
+				cachedStore.GetCache().Add("tenant-tenant123", storeId)
+
+				client.EXPECT().
+					ReadAuthorizationModels(ctx, &openfgav1.ReadAuthorizationModelsRequest{StoreId: "store123"}).
+					Return(nil, assert.AnError).
+					Once()
+			},
+			expectedError: assert.AnError,
+		},
+		{
+			name: "NoReadAuthorizationModels_Error",
+			setupMock: func(client *mocks.OpenFGAServiceClient, cachedStore *CachedFgaTenantStore) {
+				cachedStore.GetCache().Add("tenant-tenant123", storeId)
+
+				client.EXPECT().
+					ReadAuthorizationModels(ctx, &openfgav1.ReadAuthorizationModelsRequest{StoreId: "store123"}).
+					Return(&openfgav1.ReadAuthorizationModelsResponse{}, nil).
+					Once()
+			},
+			expectedError: errors.New("no authorization models in response. Cannot determine proper AuthorizationModelId"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := &mocks.OpenFGAServiceClient{}
+			cachedStore := NewCachedFgaHelper()
+			tt.setupMock(client, cachedStore)
+
+			modelID, err := cachedStore.GetModelIDForTenant(ctx, client, tenantID)
+
+			assert.Equal(t, tt.expectedModelID, modelID)
+			assert.Equal(t, tt.expectedError, err)
+
+			client.AssertExpectations(t)
+		})
+	}
+}
+
+func TestIsDuplicateWriteError(t *testing.T) {
+	tests := []struct {
+		name     string
+		err      error
+		expected bool
+	}{
+		{
+			name:     "NoError",
+			err:      nil,
+			expected: false,
+		},
+		{
+			name:     "NonGRPCError",
+			err:      errors.New("some error"),
+			expected: false,
+		},
+		{
+			name:     "NonDuplicateWriteGRPCError",
+			err:      status.Error(codes.InvalidArgument, "invalid argument"),
+			expected: false,
+		},
+		//{
```

Comment:
commented out test case, lets remove it or uncomment.

Reply:
done